### PR TITLE
add paddle jna

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##

Add paddle sample code

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- [x] Code is well-documented: 
    - For user-facing API changes, Java doc has been updated. 
    - For new examples, README.md is added to explain the what the example does.
- [x] To the my best knowledge, [examples](https://github.com/awslabs/djl/tree/master/examples) and [jupyter notebooks](https://github.com/awslabs/djl/tree/master/jupyter) are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

File: paddlepaddle/paddlepaddle-engine/src/main/java/ai/djl/paddlepaddle/jna/PD_Buffer.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+package ai.djl.paddlepaddle.jna;
+
+import com.sun.jna.Pointer;
+import com.sun.jna.Structure;
+
+/** The buffer structure defined by Paddle. */
+@SuppressWarnings({"missingjavadocmethod", "typename"})
+public class PD_Buffer extends Structure {
```
Comment (stu1130) [inline paddlepaddle/paddlepaddle-engine/src/main/java/ai/djl/paddlepaddle/jna/PD_Buffer.java:21]: nit: 
```suggestion
public class PDBuffer extends Structure {
```

File: paddlepaddle/paddlepaddle-engine/src/main/java/ai/djl/paddlepaddle/jna/PD_Buffer.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    Pointer data;
```
Comment (frankfliu) [inline paddlepaddle/paddlepaddle-engine/src/main/java/ai/djl/paddlepaddle/jna/PD_Buffer.java:22]: struct members are public

File: paddlepaddle/paddlepaddle-engine/src/main/java/ai/djl/paddlepaddle/jna/PaddleLibrary.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+/** The Paddle JNA library. */
+@SuppressWarnings({"missingjavadocmethod", "methodname"})
+public interface PaddleLibrary extends Library {
+    String PD_GetPaddleTensorName(Pointer tensor);
+
+    void PD_SetPaddleTensorShape(Pointer tensor, int[] shape, int size);
+
+    int PD_GetPaddleTensorDType(Pointer tensor);
+
+    PD_Buffer PD_GetPaddleTensorData(Pointer tensor);
```
Comment (frankfliu) [inline paddlepaddle/paddlepaddle-engine/src/main/java/ai/djl/paddlepaddle/jna/PaddleLibrary.java:28]:
```suggestion
    Pointer PD_GetPaddleTensorData(Pointer tensor);
```

File: paddlepaddle/paddlepaddle-engine/build.gradle [hunk 4]
```diff
@@ -7,6 +7,19 @@ repositories {
     }
 }
 
+processResources {
```
Comment (frankfliu) [inline paddlepaddle/paddlepaddle-engine/build.gradle:10]: We don't need download the header file.

File: paddlepaddle/paddlepaddle-engine/src/main/java/ai/djl/paddlepaddle/jna/PD_Buffer.java (hunk 1 above)
Comment (frankfliu) [reply paddlepaddle/paddlepaddle-engine/src/main/java/ai/djl/paddlepaddle/jna/PD_Buffer.java:21]: I prefer keep PD_Buffer

Comment (lanking520) [reply paddlepaddle/paddlepaddle-engine/src/main/java/ai/djl/paddlepaddle/jna/PD_Buffer.java:21]: This have to match right? Or JNA will not map this to the right structure?

Comment (frankfliu) [reply paddlepaddle/paddlepaddle-engine/src/main/java/ai/djl/paddlepaddle/jna/PD_Buffer.java:21]: I doesn't have to match the C++ name, but just make it easy to keep them the same for reference

**Review by frankfliu — APPROVED** [review]

File: paddlepaddle/paddlepaddle-engine/src/main/java/ai/djl/paddlepaddle/jna/JnaUtils.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+ */
+package ai.djl.paddlepaddle.jna;
+
+import ai.djl.ndarray.types.DataType;
+import ai.djl.paddlepaddle.engine.PpDataType;
+import com.sun.jna.Native;
+import com.sun.jna.Pointer;
+
+/**
+ * A class containing utilities to interact with the MXNet Engine's Java Native Access (JNA) layer.
```
Comment (Jzow) [inline paddlepaddle/paddlepaddle-engine/src/main/java/ai/djl/paddlepaddle/jna/JnaUtils.java:21]: MXNet Engine's Update PaddlePaddle Engine

Comment (Jzow) [inline paddlepaddle/paddlepaddle-engine/src/main/java/ai/djl/paddlepaddle/jna/JnaUtils.java:21]: MXNet Engine's Update PaddlePaddle Engine's

File: paddlepaddle/paddlepaddle-engine/src/main/java/ai/djl/paddlepaddle/jna/package-info.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+/** Contains classes to interface with the underlying MXNet Engine. */
```
Comment (Jzow) [inline paddlepaddle/paddlepaddle-engine/src/main/java/ai/djl/paddlepaddle/jna/package-info.java:14]: MXNet Engine's Update PaddlePaddle Engine's

**Pull request merged**